packages/neon-core/src/wallet/Wallet.ts

```diff
@@ -98,78 +98,58 @@ export class Wallet {
    * Attempts to decrypt Account at index in array.
    * @param index Index of Account in array.
    * @param keyphrase keyphrase
-   * @return Decryption success/failure
+   * @return Promise of decryption success/failure.
    */
-  public decrypt(index: number, keyphrase: string): boolean {
+  public async decrypt(index: number, keyphrase: string): Promise<boolean> {
     if (index < 0) {
-      throw new Error("Index cannot be negative!");
+      return Promise.reject(`Index cannot be negative! index: ${index}`);
     }
     if (index >= this.accounts.length) {
-      throw new Error("Index cannot larger than Accounts array!");
-    }
-    try {
-      this.accounts[index].decrypt(keyphrase, this.scrypt);
-      return true;
-    } catch (err) {
-      return false;
+      return Promise.reject(
+        `Index cannot larger than Accounts array! index: ${index}`
+      );
     }
+    await this.accounts[index].decrypt(keyphrase, this.scrypt);
+    return true;
   }
 
   /**
    * Attempts to decrypt all accounts with keyphrase.
    * @param keyphrase
-   * @return Each boolean represents if that Account has been decrypted successfully.
+   * @return Promise of accounts decryption success/failure .
    */
-  public decryptAll(keyphrase: string): boolean[] {
-    const results: boolean[] = [];
-    this.accounts.map((acct, i) => {
-      results.push(this.decrypt(i, keyphrase));
-    });
-    log.info(
-      `decryptAll for Wallet ${this.name}: ${results.reduce((c, p) => {
-        return p + (c ? "1" : "0");

```

Comment:
can we retain the log statement

Reply:
That won't be necessary. Promise reject error info will reveal which index goes wrong. The old log may mislead user to regard it as success even if it fails for some index.